src/lib.rs

```diff
@@ -541,21 +541,19 @@ impl ConfiguredOidc {
                 // Create authorization state from token response
                 let authorization_state = serde_json::from_slice::<AuthorizationState>(&body).unwrap(); // TODO: Idiomatically handle the error
 
+                // Add authorization state to session
+                session.authorization_state = Some(authorization_state);
+
                 // Create new session
-                let new_session = cookie::Session{
-                    authorization_state: Some(authorization_state),
-                    original_path: session.original_path.clone(),
-                    code_verifier: session.code_verifier.clone(),
-                    state: session.state.clone(),
-                }.encrypt_and_encode(self.cipher.clone(), encoded_nonce);
+                let new_session = session.clone().encrypt_and_encode(self.cipher.clone(), encoded_nonce);
```

Comment:
You can avoid the `clone` if you change `encrypt_and_encode` to take `&self` instead of `self`.